src/AST/CommonAST.cs

```diff
@@ -142,6 +144,62 @@ public enum SpanFilterCombination
     All
 }
 
+/// <summary>
+/// Project operation node representing both KQL's project operator and TraceQL's select operation
+/// Projects specified fields/columns from the data, optionally with aliases
+/// 
+/// Level 1 Support (Current):
+/// - Simple field references
+/// - Aliasing  
+/// - Basic arithmetic operations (+, -, *, /)
+/// - Simple function calls (string, math, conversion)
+/// 
+/// Level 2 Support (Future TODO):
+/// - Complex nested expressions
+/// - Conditional expressions (case)
+/// - Advanced function calls
+/// </summary>
+public class ProjectNode : OperationNode
+{
+    public override NodeKind NodeKind => NodeKind.Project;
+    
+    /// <summary>
+    /// List of field projections to include in the output
+    /// No wildcard (*) support - if all fields needed, project operation should be omitted
+    /// </summary>
+    public List<ProjectionExpression> Projections { get; set; } = new List<ProjectionExpression>();
+    
+    /// <summary>
+    /// Keyword for language-specific syntax ('project' for KQL, 'select' for TraceQL)
+    /// </summary>
+    public string? Keyword { get; set; }
+}
+
+/// <summary>
+/// Represents a single field projection with optional alias and type information
+/// </summary>
+public class ProjectionExpression : ASTNode
+{
+    public override NodeKind NodeKind => NodeKind.ProjectionExpression;
+    
+    /// <summary>
+    /// The expression to project (field name, calculation, function call, etc.)
+    /// </summary>
+    public required Expression Expression { get; set; }
+    
+    /// <summary>
+    /// Optional alias for the projected field
+    /// </summary>
+    public string? Alias { get; set; }
+    
+    /// <summary>
+    /// Optional result type of this projection expression
+    /// Required ONLY for calculated fields and function calls where type changes
+    /// Simple field selections and renames don't need this - EE engine can infer from schema
+    /// </summary>
+    public ExpressionType? ResultType { get; set; }
```

Comment:
Why not use LiteralKind there? All the return types should be (at least for now) the simple types we have defined. And once we add more complex types we want those to be supported everywhere as well.